eng/pipelines/aggregate-reports.yml

```diff
@@ -8,177 +8,205 @@ pr:
+          condition: succeededOrFailed()
+
+
+        - publish: $(Build.ArtifactStagingDirectory)
+          condition: succeededOrFailed()
+          displayName: 'Publish Report Artifacts'
+          artifact: reports
+
+        - template: /eng/pipelines/templates/steps/upload-repository-on-failure.yml
+
+      - job: Compliance
+        displayName: Analyze (no build)
+        timeoutInMinutes: 120
+        variables:
+          - template: /eng/pipelines/templates/variables/globals.yml
+        pool:
+          # CredScan build task is Windows only (https://aka.ms/credscan).
+          vmImage: 'windows-2019'
+        steps:
+        - task: ms.vss-governance-buildtask.governance-build-task-component-detection.ComponentGovernanceComponentDetection@0
+          # ComponentGovernance is currently unable to run on pull requests of public projects. Running on non-PR
+          # builds should be sufficient.
+          # Only run if building all libraries succeeded.
+          condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
+          displayName: 'Component Detection'
+
+        # Run link verification at the end as it will run no matter the pipeline passing or failing.
+        # If this task fails with an error it shouldn't prevent aggregate reports from being generated.
+        - template: /eng/common/pipelines/templates/steps/verify-links.yml
+          parameters:
+            Directory: ""
+            CheckLinkGuidance: $true
+            Condition: succeededOrFailed()
+
+  - stage: ComplianceTools
+    displayName: Compliance Tools
+    dependsOn: []
+
+    jobs:
+      - job: ComplianceTools
+        timeoutInMinutes: 120
+
+        pool:
+          name: azsdk-pool-mms-win-2019-general
+          vmImage: MMS2019
+
+        steps:
+          - template: /eng/common/pipelines/templates/steps/policheck.yml
+            parameters:
+              ExclusionDataBaseFileName: "JavaPoliCheckExclusion"
+              PublishAnalysisLogs: false
+          
+          - task: securedevelopmentteam.vss-secure-development-tools.build-task-credscan.CredScan@3
+            displayName: 'Run CredScan'
+            inputs:
+              suppressionsFile: 'eng\CredScanSuppression.json'
+            condition: succeededOrFailed()
+
+          - task: securedevelopmentteam.vss-secure-development-tools.build-task-postanalysis.PostAnalysis@2
+            displayName: 'Post Analysis'
+            inputs:
+              GdnBreakAllTools: false
+              GdnBreakGdnToolCredScan: true
+              GdnBreakGdnToolCredScanSeverity: Error
+              GdnBreakBaselineFiles: $(Build.SourcesDirectory)\eng\java.gdnbaselines
+              GdnBreakBaselines: baseline
+              # Used for generating baseline file.
+              # GdnBreakOutputBaselineFile: java
+              # GdnBreakOutputBaseline: baseline
+            condition: succeededOrFailed()
+            continueOnError: true
+
+          - pwsh: |
```

Comment:
Orthogonal to this PR. @sima-zhu is this conversion step still interesting now that we have gotten clean? Should we remove this?